modules/oe_translation_poetry/tests/Kernel/ContactFieldsTest.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_translation_poetry\Kernel;
+
+use Drupal\field\Entity\FieldConfig;
+use Drupal\field\Entity\FieldStorageConfig;
+use Drupal\Tests\oe_translation\Kernel\TranslationKernelTestBase;
+
+/**
+ * Tests the Poetry contact fields type and formatter.
+ */
+class ContactFieldsTest extends TranslationKernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'oe_translation_poetry_html_formatter',
+    'oe_translation_poetry',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->installSchema('node', ['node_access']);
+
+    $this->container->get('entity_type.manager')->getStorage('node_type')->create([
+      'type' => 'page',
+      'name' => 'Page',
+    ])->save();
+
+    FieldStorageConfig::create([
+      'field_name' => 'organisational_contact_info',
+      'entity_type' => 'node',
+      'type' => 'oe_translation_poetry_organisation_contact',
+    ])->save();
+
+    FieldConfig::create([
+      'field_name' => 'organisational_contact_info',
+      'entity_type' => 'node',
+      'bundle' => 'page',
+    ])->save();
+
+    FieldStorageConfig::create([
+      'field_name' => 'personal_contact_info',
+      'entity_type' => 'node',
+      'type' => 'oe_translation_poetry_personal_contact',
+    ])->save();
+
+    FieldConfig::create([
+      'field_name' => 'personal_contact_info',
+      'entity_type' => 'node',
+      'bundle' => 'page',
+    ])->save();
+  }
+
+  /**
+   * Tests the Organisational field type and formatter.
+   */
+  public function testOrganisationalContactItemField(): void {
+    /** @var \Drupal\node\NodeStorageInterface $node_storage */
+    $node_storage = $this->container->get('entity_type.manager')->getStorage('node');
+    $tests = [
```

Comment:
This seems like it should be handled by a data provider method.

Reply:
No, we don't wanna run new tests for each of the case. No data provider in non-Unit tests.